examples/big_int/src/main.rs

```diff
@@ -0,0 +1,102 @@
+
+#![no_std]
+#![allow(incomplete_features)]
+#![feature(allocator_api)]
+#![feature(generic_const_exprs)]
+#![no_main]
+
+// use primitive_types::U256;
+// This example shows how to use the big int library to perform arithmetic operations on large numbers.
+
+use riscv_common::{csr_read_word, zksync_os_finish_success};
+
+extern "C" {
+    // Boundaries of the heap
+    static mut _sheap: usize;
+    static mut _eheap: usize;
+
+    // Boundaries of the stack
+    static mut _sstack: usize;
+    static mut _estack: usize;
+
+    // Boundaries of the data region - to init .data section. Yet unused
+    static mut _sdata: usize;
+    static mut _edata: usize;
+    static mut _sidata: usize;
+}
+
+core::arch::global_asm!(include_str!("../../scripts/asm/asm_reduced.S"));
+
+#[no_mangle]
+extern "C" fn eh_personality() {}
+
+#[link_section = ".init.rust"]
+#[export_name = "_start_rust"]
+unsafe extern "C" fn start_rust() -> ! {
+    main()
+}
+
+#[export_name = "_setup_interrupts"]
+pub unsafe fn custom_setup_interrupts() {
+    extern "C" {
+        fn _machine_start_trap();
+    }
+
+    // xtvec::write(_machine_start_trap as *const () as usize, xTrapMode::Direct);
+}
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct MachineTrapFrame {
+    pub registers: [u32; 32],
+}
+
+/// Exception (trap) handler in rust.
+/// Called from the asm/asm.S
+#[link_section = ".trap.rust"]
+#[export_name = "_machine_start_trap_rust"]
+pub extern "C" fn machine_start_trap_rust(_trap_frame: *mut MachineTrapFrame) -> usize {
+    {
+        unsafe { core::hint::unreachable_unchecked() }
+    }
+}
+
+#[inline(always)]
+fn csr_trigger_delegation(
+    input_a: *const u32,
+    input_b: *const u32,
+    round_mask: *mut u32,
+) {
+    unsafe {
+        core::arch::asm!(
+            "csrrw x0, 0x7ca, x0",
+            in("x10") input_a.addr(),
+            in("x11") input_b.addr(),
+            inlateout("x12") round_mask,
+            options(nostack, preserves_flags)
+        )
+    }
+}
+
+#[repr(C)]
+#[repr(align(32))]
+pub struct U256(pub [u32; 4]);
+
+
+const MODULUS: u32 = 1_000_000_000;
+
+
+unsafe fn workload() -> ! {
+    let mut a = U256([1, 1, 1, 1]);
+    let b = U256([1, 1, 1, 1]);
+    let mut round_mask = 1;
+    csr_trigger_delegation( a.0.as_ptr(), b.0.as_ptr(), round_mask );
+
+
+    zksync_os_finish_success(&[round_mask, 0, 0, 0, 0, 0, 0, 0]);
```

Comment:
Imo we should output the result as well (`a[0], a[1], a[2], a[3]`) to sanity check what it did.